Sources/SpyBug/Views/ReportOptionsView.swift

```diff
@@ -7,22 +7,55 @@
 
 import SwiftUI
 
-struct ReportOptionsView: View {
+public struct ReportOptionsView: View {
+    @Environment(\.dismiss) private var dismiss
     @Environment(\.openURL) private var openURL
+#if os(visionOS)
+    @Environment(\.dismissWindow) private var dismissWindow
+    @Environment(\.openWindow) private var openWindow
+#endif
     var author: String?
     @State private var selectedType: ReportType?
     @State private var showReportForm = false
     var reportTypes: [ReportType]
     
-    var body: some View {
+    public init(showReportForm: Bool = false, author: String? = nil, reportTypes: [ReportType] = ReportType.allCases) {
+        self.showReportForm = showReportForm
+        self.author = author
+        self.reportTypes = reportTypes
+    }
+    
+    public var body: some View {
         VStack {
             if !showReportForm {
                 VStack(spacing: 16) {
-                    Text("Need help?", bundle: .module)
-                        .font(.system(size: 24, weight: .bold))
-                        .foregroundStyle(Color(.title))
-                        .padding(.vertical, 10)
-                    
+                    HStack(alignment: .center) {
+#if os(visionOS)
+                        Button {
+                            KeyboardUtils.hideKeyboard()
+                            withAnimation(.easeInOut(duration: 0.3)) {
+                                dismiss()
+                            }
+                        } label: {
+                            Image(systemName: "xmark")
+                                .font(.system(size: 28, weight: .regular))
+                                .foregroundStyle(Color(.secondary))
+                                .padding(.leading)
+                        }
+                        .buttonStyle(.plain)
+#endif
+                        Spacer()
+                        
+                        Text("Need help?", bundle: .module)
+                            .font(.system(size: 24, weight: .bold))
+                            .foregroundStyle(Color(.title))
+                            .padding(.vertical, 10)
+                        
+                        Spacer()
+                    }
+#if os(visionOS)
+                    .padding(.top, 10)
+#endif
```

Comment:
As explained in my video here 2 different ViewBuilder (one for iOS and one for VisionOS) will be clearer and easier to understand. especially if in the future we bring iPad design and MacOS

Reply:
Done